lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -5340,25 +5359,25 @@ Value *SROA_Parameter_HLSL::castArgumentIfRequired(
   return V;
 }
 
+struct AnnotatedValue {
```

Comment:
This supersedes the previous `annotationMap` in a cleaner way. It's helped me catch some bugs, too.